gemrb/core/Map.cpp

```diff
@@ -2454,9 +2461,10 @@ SearchmapPoint Map::FindFarthest(const NavmapPoint &d, unsigned int size, unsign
 	while (!open.empty()) {
 		smptCurrent = open.top().point;
 		open.pop();
+		int sgn = (RAND(0, 1)) ? -1 : 1;
```

Comment:
I suggest you add a comment, eg. "add some noise to the position, so infinte loops are less likely"